From e9d7894ba677903f0a2973919dc87aece206f000 Mon Sep 17 00:00:00 2001
From: Sandro Tosi <morph@debian.org>
Date: Wed, 4 Nov 2015 03:21:15 +0000
Subject: dont-remove-build-dir

---
 setup.py | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/setup.py b/setup.py
index 1e1e220..cb3274c 100644
--- a/setup.py
+++ b/setup.py
@@ -22,11 +22,6 @@ Programming Language :: Python :: 3
 Programming Language :: Python :: 3.2
 """.splitlines()))
 
-try:
-    shutil.rmtree("./build")
-except(OSError):
-    pass
-
 module1 = Extension('ujson',
                     sources = ['./python/ujson.c', 
                                './python/objToJSON.c', 
