From b0935cc7181b96ab050ba2c6f6a3f355bf6086e0 Mon Sep 17 00:00:00 2001
From: Sandro Tosi <morph@debian.org>
Date: Sun, 17 Jan 2016 23:02:39 +0000
Subject: run test more verbosely

---
 tests/tests.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/tests.py b/tests/tests.py
index 23db199..3ed6ee5 100644
--- a/tests/tests.py
+++ b/tests/tests.py
@@ -956,7 +956,7 @@ raise NotImplementedError("Implement this test!")
 """
 
 if __name__ == "__main__":
-    unittest.main()
+    unittest.main(verbosity=2)
 
 """
 # Use this to look for memory leaks
