diff -urNp a/agent/mibgroup/disman/event/mteTrigger.c b/agent/mibgroup/disman/event/mteTrigger.c
--- a/agent/mibgroup/disman/event/mteTrigger.c	2018-09-27 10:43:38.722444233 +0200
+++ b/agent/mibgroup/disman/event/mteTrigger.c	2018-09-27 11:01:46.503253963 +0200
@@ -1012,7 +1012,7 @@ mteTrigger_run( unsigned int reg, void *
                      * Similarly, if no fallEvent is configured,
                      *  there's no point in trying to fire it either.
                      */
-                    if (entry->mteTThRiseEvent[0] != '\0' ) {
+                    if (entry->mteTThFallEvent[0] != '\0' ) {
                         entry->mteTriggerXOwner   = entry->mteTThObjOwner;
                         entry->mteTriggerXObjects = entry->mteTThObjects;
                         entry->mteTriggerFired    = vp1;
@@ -1105,7 +1105,7 @@ mteTrigger_run( unsigned int reg, void *
                      * Similarly, if no fallEvent is configured,
                      *  there's no point in trying to fire it either.
                      */
-                    if (entry->mteTThDRiseEvent[0] != '\0' ) {
+                    if (entry->mteTThDFallEvent[0] != '\0' ) {
                         entry->mteTriggerXOwner   = entry->mteTThObjOwner;
                         entry->mteTriggerXObjects = entry->mteTThObjects;
                         entry->mteTriggerFired    = vp1;
diff -urNp a/agent/mibgroup/hardware/cpu/cpu_linux.c b/agent/mibgroup/hardware/cpu/cpu_linux.c
--- a/agent/mibgroup/hardware/cpu/cpu_linux.c	2018-09-27 10:43:38.697444449 +0200
+++ b/agent/mibgroup/hardware/cpu/cpu_linux.c	2018-09-27 11:12:07.109024625 +0200
@@ -122,6 +122,7 @@ int netsnmp_cpu_arch_load( netsnmp_cache
         bsize = getpagesize()-1;
         buff = (char*)malloc(bsize+1);
         if (buff == NULL) {
+            close(statfd);
             return -1;
         }
     }
diff -urNp a/agent/mibgroup/ip-mib/data_access/ipaddress_linux.c b/agent/mibgroup/ip-mib/data_access/ipaddress_linux.c
--- a/agent/mibgroup/ip-mib/data_access/ipaddress_linux.c	2018-09-27 10:43:38.711444328 +0200
+++ b/agent/mibgroup/ip-mib/data_access/ipaddress_linux.c	2018-09-27 11:16:45.532231535 +0200
@@ -543,15 +543,18 @@ netsnmp_access_ipaddress_extra_prefix_in
     status = send (sd, &req, req.nlhdr.nlmsg_len, 0);
     if (status < 0) {
         snmp_log(LOG_ERR, "could not send netlink request\n");
+        close(sd);
         return -1;
     }
     status = recv (sd, buf, sizeof(buf), 0);
     if (status < 0) {
         snmp_log (LOG_ERR, "could not recieve netlink request\n");
+        close(sd);
         return -1;
     }
     if (status == 0) {
        snmp_log (LOG_ERR, "nothing to read\n");
+       close(sd);
        return -1;
     }
     for (nlmp = (struct nlmsghdr *)buf; status > sizeof(*nlmp); ){
@@ -561,11 +564,13 @@ netsnmp_access_ipaddress_extra_prefix_in
 
         if (req_len < 0 || len > status) {
             snmp_log (LOG_ERR, "invalid netlink message\n");
+            close(sd);
             return -1;
         }
 
         if (!NLMSG_OK (nlmp, status)) {
             snmp_log (LOG_ERR, "invalid NLMSG message\n");
+            close(sd);
             return -1;
         }
         rtmp = (struct ifaddrmsg *)NLMSG_DATA(nlmp);
