diff -up boost_1_53_0/boost/mpi/datatype.hpp\~ boost_1_53_0/boost/mpi/datatype.hpp
--- boost_1_53_0/boost/mpi/datatype.hpp~	2010-10-18 08:22:09.000000000 +0200
+++ boost_1_53_0/boost/mpi/datatype.hpp	2014-02-19 03:14:48.564100698 +0100
@@ -332,7 +332,7 @@ struct is_mpi_datatype<bool>
 #ifndef BOOST_MPI_DOXYGEN
 // direct support for special primitive data types of the serialization library
 BOOST_MPI_DATATYPE(boost::archive::library_version_type, get_mpi_datatype(uint_least16_t()), integer);
-BOOST_MPI_DATATYPE(boost::archive::version_type, get_mpi_datatype(uint_least8_t()), integer);
+BOOST_MPI_DATATYPE(boost::archive::version_type, get_mpi_datatype(uint_least32_t()), integer);
 BOOST_MPI_DATATYPE(boost::archive::class_id_type, get_mpi_datatype(int_least16_t()), integer);
 BOOST_MPI_DATATYPE(boost::archive::class_id_reference_type, get_mpi_datatype(int_least16_t()), integer);
 BOOST_MPI_DATATYPE(boost::archive::class_id_optional_type, get_mpi_datatype(int_least16_t()), integer);
@@ -340,7 +340,7 @@ BOOST_MPI_DATATYPE(boost::archive::objec
 BOOST_MPI_DATATYPE(boost::archive::object_reference_type, get_mpi_datatype(uint_least32_t()), integer);
 BOOST_MPI_DATATYPE(boost::archive::tracking_type, get_mpi_datatype(bool()), builtin);
 BOOST_MPI_DATATYPE(boost::serialization::collection_size_type, get_mpi_datatype(std::size_t()), integer);
-BOOST_MPI_DATATYPE(boost::serialization::item_version_type, get_mpi_datatype(uint_least8_t()), integer);
+BOOST_MPI_DATATYPE(boost::serialization::item_version_type, get_mpi_datatype(unsigned()), integer);
 #endif // Doxygen
 
 

Diff finished.  Wed Feb 19 03:14:54 2014
