--- boost_1_53_0/tools/build/v2/tools/python.jam.orig	2012-04-26 04:35:55.000000000 +0100
+++ boost_1_53_0/tools/build/v2/tools/python.jam	2016-05-11 15:13:47.628380504 +0100
@@ -95,7 +95,7 @@ feature.feature pythonpath : : free opti
 #   using python : 2.3 : /usr/local/bin/python ;
 #
 rule init ( version ? : cmd-or-prefix ? : includes * : libraries ?
-    : condition * : extension-suffix ? )
+    : condition * : extension-suffix ? : abi-letters ? )
 {
     project.push-current $(.project) ;
 
@@ -108,7 +108,7 @@ rule init ( version ? : cmd-or-prefix ?
         }
     }
 
-    configure $(version) : $(cmd-or-prefix) : $(includes) : $(libraries) : $(condition) : $(extension-suffix) ;
+    configure $(version) : $(cmd-or-prefix) : $(includes) : $(libraries) : $(condition) : $(extension-suffix) : $(abi-letters) ;
 
     project.pop-current ;
 }
@@ -662,7 +662,7 @@ local rule system-library-dependencies (
 
 # Declare a target to represent Python's library.
 #
-local rule declare-libpython-target ( version ? : requirements * )
+local rule declare-libpython-target ( version ? : requirements * : abi-letters ? )
 {
     # Compute the representation of Python version in the name of Python's
     # library file.
@@ -686,13 +686,13 @@ local rule declare-libpython-target ( ve
     }
 
     # Declare it.
-    lib python.lib : : <name>python$(lib-version) $(requirements) ;
+    lib python.lib : : <name>python$(lib-version)$(abi-letters) $(requirements) ;
 }
 
 
 # Implementation of init.
 local rule configure ( version ? : cmd-or-prefix ? : includes * : libraries ? :
-    condition * : extension-suffix ? )
+    condition * : extension-suffix ? : abi-letters ? )
 {
     local prefix ;
     local exec-prefix ;
@@ -708,6 +708,7 @@ local rule configure ( version ? : cmd-o
         extension-suffix ?= _d ;
     }
     extension-suffix ?= "" ;
+    abi-letters ?= "" ;
 
     # Normalize and dissect any version number.
     local major-minor ;
@@ -931,7 +932,7 @@ local rule configure ( version ? : cmd-o
     }
     else
     {
-        declare-libpython-target $(version) : $(target-requirements) ;
+        declare-libpython-target $(version) : $(target-requirements) : $(abi-letters) ;
 
         # This is an evil hack.  On, Windows, when Python is embedded, nothing
         # seems to set up sys.path to include Python's standard library
