diff -up boost_1_53_0/tools/build/v2/tools/gcc.jam\~ boost_1_53_0/tools/build/v2/tools/gcc.jam
--- boost_1_53_0/tools/build/v2/tools/gcc.jam~	2012-04-26 05:35:55.000000000 +0200
+++ boost_1_53_0/tools/build/v2/tools/gcc.jam	2014-02-28 16:32:22.575212631 +0100
@@ -363,24 +363,6 @@ generators.override gcc.compile.c++.pch 
 
 toolset.flags gcc.compile PCH_FILE <pch>on : <pch-file> ;
 
-# Declare flags and action for compilation.
-toolset.flags gcc.compile OPTIONS <optimization>off   : -O0 ;
-toolset.flags gcc.compile OPTIONS <optimization>speed : -O3 ;
-toolset.flags gcc.compile OPTIONS <optimization>space : -Os ;
-
-toolset.flags gcc.compile OPTIONS <inlining>off  : -fno-inline ;
-toolset.flags gcc.compile OPTIONS <inlining>on   : -Wno-inline ;
-toolset.flags gcc.compile OPTIONS <inlining>full : -finline-functions -Wno-inline ;
-
-toolset.flags gcc.compile OPTIONS <warnings>off : -w ;
-toolset.flags gcc.compile OPTIONS <warnings>on  : -Wall ;
-toolset.flags gcc.compile OPTIONS <warnings>all : -Wall -pedantic ;
-toolset.flags gcc.compile OPTIONS <warnings-as-errors>on : -Werror ;
-
-toolset.flags gcc.compile OPTIONS <debug-symbols>on : -g ;
-toolset.flags gcc.compile OPTIONS <profiling>on : -pg ;
-toolset.flags gcc.compile OPTIONS <rtti>off : -fno-rtti ;
-
 rule setup-fpic ( targets * : sources * : properties * )
 {
     local link = [ feature.get-values link : $(properties) ] ;

Diff finished.  Fri Feb 28 16:32:24 2014
