--- boost_1_53_0/tools/build/v2/tools/python.jam.orig	2016-05-11 15:13:47.628380504 +0100
+++ boost_1_53_0/tools/build/v2/tools/python.jam	2016-05-11 15:14:25.337485463 +0100
@@ -996,7 +996,7 @@ local rule configure ( version ? : cmd-o
     else
     {
         alias python_for_extensions
-            :
+            : python
             : $(target-requirements)
             :
             : $(usage-requirements)
