--- ez_setup.py	2018-05-22 10:27:30.779776894 +0200
+++ ez_setup.py	2018-05-22 10:27:16.739684276 +0200
@@ -47,7 +47,7 @@
         return os.spawnl(os.P_WAIT, sys.executable, *args) == 0

 DEFAULT_VERSION = "0.6.14"
-DEFAULT_URL = "http://pypi.python.org/packages/source/d/distribute/"
+DEFAULT_URL = "https://pypi.python.org/packages/source/d/distribute/"
 SETUPTOOLS_FAKED_VERSION = "0.6c11"

 SETUPTOOLS_PKG_INFO = """\
