Description: Patch in download link for a Juju environment file.
Author: James Page <james.page@ubuntu.com>
Forwarded: not-needed

--- /dev/null
+++ b/openstack_dashboard/dashboards/project/access_and_security/templates/access_and_security/api_access/juju.environments.template
@@ -0,0 +1,22 @@
+# To use Juju with your Openstack cloud:
+#    1. Run: apt-get install juju
+#    2. Copy this file to $HOME/.juju/environments.yaml
+#    3. Complete any empty fields
+#    4. Run 'juju bootstrap' and deploy services!
+#
+# To find out more about juju visit http://juju.ubuntu.com
+environments:
+  openstack:
+    type: openstack
+    auth-url: {{ auth_url }}
+    auth-mode: userpass
+    use-floating-ip: true
+    # Please provide the UUID of the network to use if you have
+    # more than one network configured for your tenant.
+    # network:
+    region: {{ region }}
+    username: {{ user }}
+    # Please provide the password you use to access the
+    # OpenStack cloud.
+    password:
+    tenant-name: {{ tenant_name }}
--- a/openstack_dashboard/dashboards/project/access_and_security/api_access/views.py
+++ b/openstack_dashboard/dashboards/project/access_and_security/api_access/views.py
@@ -151,3 +151,37 @@ class CredentialsView(forms.ModalFormMix
                 exceptions.handle(self.request,
                                   _('Unable to get EC2 credentials'))
         return context
+
+
+def download_juju_env_file(request):
+    tenant_id = request.user.tenant_id
+    tenant_name = request.user.tenant_name
+    region = request.user.services_region
+    if region is None:
+        region = ""
+
+    template = 'project/access_and_security/api_access/juju.environments.template'
+
+    try:
+        keystone_url = api.base.url_for(request,
+                                        'identity',
+                                        endpoint_type='publicURL')
+        context = {'auth_url': keystone_url,
+                   'tenant_name': tenant_name,
+                   'user': request.user,
+                   'region': region}
+
+        response = shortcuts.render(request,
+                                    template,
+                                    context,
+                                    content_type="text/yaml")
+        response['Content-Disposition'] = ('attachment; '
+                                           'filename="%s-environments.yaml"'
+                                           % tenant_name)
+        response['Content-Length'] = str(len(response.content))
+        return response
+
+    except Exception as e:
+        LOG.exception("Exception in DownloadJujuEnvironmentsForm.")
+        messages.error(request, _('Error Downloading Juju Environments File: %s') % e)
+        return shortcuts.redirect(request.build_absolute_uri())
--- a/openstack_dashboard/dashboards/project/access_and_security/api_access/urls.py
+++ b/openstack_dashboard/dashboards/project/access_and_security/api_access/urls.py
@@ -28,5 +28,6 @@ urlpatterns = patterns(
     url(r'^ec2/$', views.download_ec2_bundle, name='ec2'),
     url(r'^openrc/$', views.download_rc_file, name='openrc'),
     url(r'^view_credentials/$', views.CredentialsView.as_view(),
-        name='view_credentials')
+        name='view_credentials'),
+    url(r'^juju/$', views.download_juju_env_file, name='juju'),
 )
--- a/openstack_dashboard/dashboards/project/access_and_security/api_access/tables.py
+++ b/openstack_dashboard/dashboards/project/access_and_security/api_access/tables.py
@@ -56,6 +56,14 @@ class ViewCredentials(tables.LinkAction)
     url = "horizon:project:access_and_security:api_access:view_credentials"
 
 
+class DownloadJujuEnv(tables.LinkAction):
+    name = "download_jujuenv"
+    verbose_name = _("Download Juju Environment File")
+    verbose_name_plural = _("Download Juju Environment File")
+    icon = "download"
+    url = "horizon:project:access_and_security:api_access:juju"
+
+
 class EndpointsTable(tables.DataTable):
     api_name = tables.Column('type',
                              verbose_name=_("Service"),
@@ -67,4 +75,4 @@ class EndpointsTable(tables.DataTable):
         name = "endpoints"
         verbose_name = _("API Endpoints")
         multi_select = False
-        table_actions = (DownloadOpenRC, DownloadEC2, ViewCredentials)
+        table_actions = (DownloadOpenRC, DownloadEC2, ViewCredentials, DownloadJujuEnv)
--- a/openstack_dashboard/usage/tables.py
+++ b/openstack_dashboard/usage/tables.py
@@ -29,6 +29,14 @@ class CSVSummary(tables.LinkAction):
         return self.table.kwargs['usage'].csv_link()
 
 
+class DownloadJujuEnv(tables.LinkAction):
+    name = "download_jujuenv"
+    verbose_name = _("Download Juju Environment File")
+    verbose_name_plural = _("Download Juju Environment File")
+    icon = "download"
+    url = "horizon:project:access_and_security:api_access:juju"
+
+
 class BaseUsageTable(tables.DataTable):
     vcpus = tables.Column('vcpus', verbose_name=_("VCPUs"))
     disk = tables.Column('local_gb', verbose_name=_("Disk"),
@@ -95,5 +103,5 @@ class ProjectUsageTable(BaseUsageTable):
         hidden_title = False
         verbose_name = _("Usage")
         columns = ("instance", "vcpus", "disk", "memory", "uptime")
-        table_actions = (CSVSummary,)
+        table_actions = (CSVSummary, DownloadJujuEnv,)
         multi_select = False
